Formula/gnvim.rb

```diff
@@ -0,0 +1,24 @@
+class Gnvim < Formula
+  desc "GUI for neovim, without any web bloat"
+  homepage "https://github.com/vhakulinen/gnvim"
+  url "https://github.com/vhakulinen/gnvim.git",
+    tag:      "v0.1.6",
+    revision: "7492a385856b2d2e142d9cbd10a4dcaf0ccc16fa"
+  license "MIT"
+
+  depends_on "rust" => :build
+  depends_on "atk"
+  depends_on "gtk+3"
+  depends_on "libepoxy"
+  depends_on "librsvg"
+
+  def install
+    system "cargo", "install", "--no-default-features", *std_cargo_args
+  end
+
+  test do
+    # gnvim doesn't support headless mode or calling vim scripts, so these are very basic
```

Comment:
You can do something like `zshdb` and provide key input. Maybe save a file?

Reply:
I'm unsure how in this case - this is a GUI wrapper for neovim and doesn't accept input on stdin - only through the GUI.

Reply:
@SMillerDev: given that I don't currently know how allow testing without running the GUI or to simulate GUI input to simulate interaction with gnvim, is there something else you would suggest for this formula?

Reply:
https://github.com/BlueM/cliclick

`brew install cliclick` and play around with it. It can be declared as a `:test` dependency.

Reply:
> https://github.com/BlueM/cliclick

* It asks me for accessibility permissions to be manually granted in order to simulate keypresses or mouse clicks.
* Would it work in a headless setting in which these tests may be run?
* A test like this seems that it would be fragile - I couldn't find a way to force a window to be in focus, I can only hope that the gnvim window is in focus after I wait for a while.
* I couldn't find any use of it in any other formula - isn't this a common issue for GUI applications? What solutions are they using?
* Checking a sample of other GUI formulas a silly check for `--version` seems to be pretty popular as the sole test

So I'm in the same spot - I'd love to write a more substantial test for gnvim, but given what gnvim supports and the constraints as I understand them I don't know how to do that.

Reply:
If it's running Neovim in the background then you should be able to control it from the command-line with neovim-remote: https://github.com/mhinz/neovim-remote

Reply:
> * It asks me for accessibility permissions to be manually granted in order to simulate keypresses or mouse clicks.

Oh right. It's been a while since I installed it.

> * Would it work in a headless setting in which these tests may be run?

`cataclysm`, for instance, is a GUI game app that was tested successfully in CI just recently, so I'd say yes. That said, its test just verifies that it populates the necessary user state when presented with a fresh sandbox home. If `gnvim` can do something similar, that might be an acceptable substitute.

> * A test like this seems that it would be fragile - I couldn't find a way to force a window to be in focus, I can only hope that the gnvim window is in focus after I wait for a while.

That might be a concern if multiple windows open during a gnvim launch, otherwise any text input should automatically go to the correct window. Only one way to find out. :grin:

> * I couldn't find any use of it in any other formula - isn't this a common issue for GUI applications? What solutions are they using?

None, as far as I know. As you've already noticed, most tests of GUI apps are version tests right now, probably because no one's really bothered to do the necessary. That there seem to be few issues reported thus far can be attributed more to dumb luck than "GUI apps don't need to be tested".

Reply:
I agree that a test exercising real logic would be excellent.

> Only one way to find out.

Agreed, I'll write something and we'll see what CI thinks of it.